frost/src/spatial.rs

```diff
@@ -0,0 +1,132 @@
+use crate::{Error, FrostObs};
+use chrono::prelude::*;
+use rstar::RTree;
+use olympian::points::{CoordinateType, Points};
+use rove::data_switch::{SpatialCache, Timestamp};
+use rove::pb::util::GeoPoint;
+
+// have to repeat from series? 
+// or is it possible to import - function `crate::series::extract_obs` exists but is inaccessible
+fn extract_obs(mut resp: serde_json::Value) -> Result<Vec<FrostObs>, Error> {
+    let obs_portion = resp
+        .get_mut("data")
+        .ok_or(Error::FindObs(
+            "couldn't find data field on root".to_string(),
+        ))?
+        .get_mut("tseries")
+        .ok_or(Error::FindObs(
+            "couldn't find tseries field on data".to_string(),
+        ))?
+        .get_mut(0)
+        .ok_or(Error::FindObs("tseries array is empty".to_string()))?
+        .get_mut("observations")
+        .ok_or(Error::FindObs(
+            "couldn't observations data field on 1st tseries".to_string(),
+        ))?
+        .take();
+
+    let obs: Vec<FrostObs> = serde_json::from_value(obs_portion)?;
+
+    Ok(obs)
+}
+
+fn json_to_spatial_cache(
+    resp: serde_json::Value,
+    polygon: Vec<GeoPoint>,
+    extra_spec: &str,
+    timestamp: DateTime<Utc>,
+) -> Result<SpatialCache, Error> {
+    let obses: Vec<FrostObs> = extract_obs(resp)?;
+
+    let mut data = Vec::new();
+    let mut tree = RTree::new();
+    let mut rtree: Points = Points{ 
+        tree: tree,
+        lats: Vec::new(),
+        lons: Vec::new(),
+        elevs: Vec::new(),
+        lafs: Vec::new(),
+        ctype: CoordinateType::Cartesian,
+     };
+
+    Ok(SpatialCache {
+        rtree,
+        data,
+    })
```

Comment:
SpatialCache has a constructor, you Don't need to make the rtree manually. See lustre_netatmo for an example